### PR TITLE
Helm3: fix list showing multiple revisions of release

### LINKED ISSUE #6356: helm 3.0 beta2 has same release in one namespace
helm install from repo and upgrade from local
```
helm install prometheus-operator  offical-repo/prometheus-operator --namespace=kube-system 
helm upgrade prometheus-operator .  -n kube-system -f values.yaml

helm list -n kube-system                                      
NAME                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                    
prometheus-operator     kube-system     16              2019-09-01 12:18:36.589258986 +0800 CST failed          prometheus-operator-6.8.0
prometheus-operator     kube-system     32              2019-09-04 20:57:28.827982696 +0800 CST deployed        prometheus-operator-6.8.0
```

Output of `helm version`:
```
version.BuildInfo{Version:"v3.0.0-beta.2", GitCommit:"26c7338408f8db593f93cd7c963ad56f67f662d4", GitTreeState:"clean", GoVersion:"go1.12.9"}
```

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:02:58Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
none

### PR DESCRIPTION
**Pull request by karuppiah7890**
Signed-off-by: Karuppiah Natarajan <karuppiah7890@gmail.com>

**What this PR does / why we need it**:
Fixes #6356 

**Special notes for your reviewer**:
Everywhere list showed up all releases (old revisions and latest revisions of releases), but in this PR, only latest revisions of releases are returned by the list methods. Let me know if list was actually intended to show both latest and old revisions and if it's needed, in which case, we can fix the issue differently after discussing a different solution 😄 

**If applicable**:
- [ ] this PR contains documentation
- [X] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by karuppiah7890** [comment]
cc @bacongobbler

File: pkg/storage/driver/secrets.go [hunk 1]
```diff
@@ -96,6 +96,13 @@ func (secrets *Secrets) List(filter func(*rspb.Release) bool) ([]*rspb.Release,
 			secrets.Log("list: failed to decode release: %v: %s", item, err)
 			continue
 		}
+		releases = append(releases, rls)
+	}
+
+	latestReleases := getLatestReleases(releases)
```
Comment (bacongobbler) [inline pkg/storage/driver/secrets.go]: This doesn't seem like it should be the responsibility of `List`. A call to `List` should be able to fetch more than just the latest releases. This should probably be refactored as a chain to the `filter` passed along to `List`. I'm going to take a look at alternatives to this approach.

**Comment by bacongobbler** [comment]
I'm going to close this in favour of #6479 as this is not the approach we want to take. There are valid cases where we want the release list to show every release in the ledger. The correct approach is to filter in the action package so business logic from `helm list` isn't leaking into other packages. Thanks for kicking this off @karuppiah7890!

**Comment by karuppiah7890** [comment]
I had doubts about the solution myself too. That's why I mentioned the solution in the issue. May be next time I'll discuss and get a 👍 on the approach then work on it. Spent too much time and effort on this only to see approach was wrong 🙈

**Pull request closed without merging**